IT tests (part 1) failing when using a custom Java with `LS_JAVA_HOME`

Noticed this while working on migrating the [Jenkins JDK matrix](https://logstash-ci.elastic.co/job/elastic-logstash-main-matrix-jdk/) job to Buildkite.

The IT job / part 1 (i.e. the ones triggered with `ci/integration_tests.sh split 0`) is failing when using a custom Java with `LS_JAVA_HOME`.
The error is (Internal [buildkite link](https://buildkite.com/elastic/logstash-jdk-matrix-pipeline/builds/15#018b9601-435a-4a21-a37d-83c82c30fb5e/2442-4104)):

```
Failures:
--
  |  
  | 1) CLI > starts without unexected warnings
  | Failure/Error: expect(up_line).to match(/^Sending Logstash logs to/)
  |  
  | expected "WARNING: Logstash comes bundled with the recommended JDK(Temurin-17.0.9+9), but is overridden by the version defined in LS_JAVA_HOME. Consider clearing LS_JAVA_HOME to use the bundled JDK." to match /^Sending Logstash logs to/
  | Diff:
  | @@ -1 +1 @@
  | -/^Sending Logstash logs to/
  | +"WARNING: Logstash comes bundled with the recommended JDK(Temurin-17.0.9+9), but is overridden by the version defined in LS_JAVA_HOME. Consider clearing LS_JAVA_HOME to use the bundled JDK."
  |  
  | # ./specs/command_line_spec.rb:43:in `block in <main>'
  | # /opt/buildkite-agent/builds/bk-agent-prod-gcp-1699028797058333096/elastic/logstash-jdk-matrix-pipeline/build/qa/integration/vendor/jruby/3.1.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in <main>'
  | # /opt/buildkite-agent/builds/bk-agent-prod-gcp-1699028797058333096/elastic/logstash-jdk-matrix-pipeline/build/qa/integration/vendor/jruby/3.1.0/gems/logstash-devutils-2.5.0-java/lib/logstash/devutils/rspec/spec_helper.rb:61:in `block in <main>'
  | # ./rspec.rb:36:in `<main>'
  |  
  | Finished in 19 minutes 10 seconds (files took 5.13 seconds to load)
  | 40 examples, 1 failure
  |  
  | Failed examples:
  |  
  | rspec ./specs/command_line_spec.rb:38 # CLI > starts without unexected warnings
  |  
  | Randomized with seed 5079
  |  
  |  
  | Gradle Test Executor 3 finished executing tests.
  |  
  | > Task :logstash-integration-tests:integrationTests FAILED
  |  
  | org.logstash.integration.RSpecTests > rspecTests FAILED
  | java.lang.AssertionError: RSpec test suit saw at least one failure.
  | at org.junit.Assert.fail(Assert.java:88)
  | at org.logstash.integration.RSpecTests.rspecTests(RSpecTests.java:79)
  |  
  | --------------------------------------------------------------------
  | \|  Results: FAILURE (1 tests, 0 successes, 1 failures, 0 skipped)  \|
  | --------------------------------------------------------------------
  |  
  | 1 test completed, 1 failed
  | Finished generating test XML results (0.029 secs) into: /opt/buildkite-agent/builds/bk-agent-prod-gcp-1699028797058333096/elastic/logstash-jdk-matrix-pipeline/qa/integration/build/test-results/integrationTests
  | Generating HTML test report...
  | Finished generating test html results (0.028 secs) into: /opt/buildkite-agent/builds/bk-agent-prod-gcp-1699028797058333096/elastic/logstash-jdk-matrix-pipeline/qa/integration/build/reports/tests/integrationTests
  | Watching 5415 directories to track changes
  | Watching 5420 directories to track changes
  | Watching 5421 directories to track changes
  | :logstash-integration-tests:integrationTests (Thread[Execution worker,5,main]) completed. Took 19 mins 23.364 secs.
  |  
  | FAILURE: Build failed with an exception.
  |  
  | * What went wrong:
  | Execution failed for task ':logstash-integration-tests:integrationTests'.
  | > There were failing tests. See the report at: file:///opt/buildkite-agent/builds/bk-agent-prod-gcp-1699028797058333096/elastic/logstash-jdk-matrix-pipeline/qa/integration/build/reports/tests/integrationTests/index.html
```

Relates: https://github.com/elastic/ingest-dev/issues/1725#event-10694112459

## Reproduction

Set:

```
export JAVA_CUSTOM_DIR=/path/to/some/custom/java
export BUILD_JAVA_HOME=$JAVA_CUSTOM_DIR
export RUNTIME_JAVA_HOME=$JAVA_CUSTOM_DIR
export LS_JAVA_HOME=$JAVA_CUSTOM_DIR
```

and from a logstash#main checkout run

```
./gradlew clean
ci/integration_tests.sh split 0
```

--- a/qa/integration/specs/command_line_spec.rb
+++ b/qa/integration/specs/command_line_spec.rb
@@ -39,7 +39,7 @@ describe "CLI >" do
     execute = @logstash.run
     lines = execute.stderr_and_stdout.split("\n")
     expect(lines.shift).to match(/^(Using system java)|(Using bundled JDK)|(Using LS_JAVA_HOME defined java):/)
-    while (up_line = lines.shift).match(/OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated|warning: ignoring JAVA_TOOL_OPTIONS|warning: already initialized constant Socket::Constants|.*warning: method redefined; discarding old to_int$|.*warning: method redefined; discarding old to_f$/) do end
+    while (up_line = lines.shift).match(/OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated|warning: ignoring JAVA_TOOL_OPTIONS|warning: already initialized constant Socket::Constants|.*warning: method redefined; discarding old to_int$|.*warning: method redefined; discarding old to_f$|^WARNING: Logstash comes bundled with the recommended JDK.*/) do end
     expect(up_line).to match(/^Sending Logstash logs to/)
   end
 end

